Setter of Access token

anshulg954 opened the pull request:
### Change Description

*Try to be precise. You can additionally add comments to your PR, this might help the reviewer a lot.*
Enhancement to set the access token if the user already has it. 
- The user can now set the token via  init(access_token="access_token"). If the access token is valid we will use it, otherwise the pre-existing prompting will continue.
- Also the user can view his/her token by get_token() directly.
- Additional Information regarding access token can also be found in all the above functionalities.
- Test cases have been updated and added.
- Readme.md has been updated as per functionality.
- Fixed Issues in resetting token. Earlier it was just deleting the file but not clearing the existing cache and was not updating the existing config.
- Also gave a temporary fix for reverify email() which has been fixed thoroughly in hashing PR.  

**If you used new dependencies: Did you add them to `requirements.txt`?**

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**
@liam-sbhoo 
## Breaking changes

If you made any breaking changes, please update the version number.
Breaking changes are totally fine, we just need to make sure to keep the users informed and the server in sync.

**Does this PR break the API? If so, what is the corresponding server commit?**
No

**Does this PR break the user interface? If so, why?**
No
---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

liam-sbhoo commented:
Thanks for the work!

I think we could add one more feature, where the user can set the token directly without having to go through the prompting. One example is, `tabpfn.init(access_token='blablabla')`.

liam-sbhoo reviewed (changes requested):
Thanks for the work!

I think we could add one more feature, where the user can set the token directly without having to go through the prompting. One example is, tabpfn.init(access_token='blablabla').

noahho commented:
This is a misunderstaning in how the access token is sued. We only want to set the access token using a functional interface, so by calling tabpfn.set_token(...) not using the prompt interface. The token is useful if you want to programatically login, so I think we don't need any new code, but only add to the README doc how to login using the token

liam-sbhoo reviewed (dismissed):
Can you add some test cases? Also perhaps a demonstration on this work.

P.S: When you create a PR, would be nice to provide the reviewer an idea/demonstration of the new workflow.

anshulg954 commented:
Hi Liam!
I did not think it as of a major change the first time I created this PR as it was a few lines of change. However, later forgot to modify the description about the changes. Sincere apologies for the same. 

As per your comments, I have updated the code and added test cases as well keeping the changes to minimal.

SamuelGabriel commented:
I think this PR does not have much new stuff compared to the current main. I would close this, ok? Sorry about this
